### PR TITLE
added compose example with JMX

### PR DESCRIPTION
**Pull request by coughman**

### CODE REVIEW

**Comment by ewencp** [comment]
@coughman who should review? core team? clients? can you assign one of the teams as reviewer?

**Comment by coughman** [comment]
assigned @aayars and clients

**Review by aayars — COMMENTED** [review]
Don't we need to set offset topic replication to 1 here?

**Comment by edenhill** [comment]
@mageshn Can you review this please?

**Comment by coughman** [comment]
@joel-hamill can you proof read the README file?

**Comment by coughman** [comment]
@aayars fixed replication factor in latest commit

File: examples/JMX/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+services:
+  zookeeper:
+    image: confluentinc/cp-zookeeper:latest
+    ports:
+      - "2181:2181"
+      - "52181:52181"
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 2181
+      ZOOKEEPER_TICK_TIME: 2000
+      KAFKA_REST_JMX_HOSTNAME: "0.0.0.0"
```
Comment (mageshn) [inline examples/JMX/docker-compose.yaml]: why KAFKA_REST in zk service

File: examples/JMX/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    ports:
+      - "9092:9092"
+      - "59092:59092"
+    environment:
+      KAFKA_ZOOKEEPER_CONNECT: "zookeeper:2181"
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092
+      KAFKA_DELETE_TOPIC_ENABLE: "true"
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
+      KAFKA_REST_JMX_HOSTNAME: "0.0.0.0"
```
Comment (mageshn) [inline examples/JMX/docker-compose.yaml]: similar to above, kafka_rest in kafka doesn't seem right

File: examples/JMX/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - "8082:8082"
+      - "58082:58082"
+    environment:
+      KAFKA_REST_ZOOKEEPER_CONNECT: "zookeeper:2181"
+      KAFKA_REST_BOOTSTRAP_SERVERS: "PLAINTEXT://kafka:9092"
+      KAFKA_REST_SCHEMA_REGISTRY_URL: "schema-registry:8081"
+      KAFKA_REST_HOST_NAME: "rest-proxy"
+      KAFKA_REST_JMX_HOSTNAME: "0.0.0.0"
+      KAFKA_REST_JMX_PORT: "58082"
+      KAFKAREST_JMX_OPTS: "-Dcom.sun.management.jmxremote=true -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.rmi.port=58082 -Dcom.sun.management.jmxremote.port=58082 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.local.only=false -Djava.rmi.server.hostname=0.0.0.0"
```
Comment (mageshn) [inline examples/JMX/docker-compose.yaml]: shouldn't KAFKAREST_JMX_OPTS be KAFKA_REST_JMX_OPTS

**Review by mageshn — CHANGES_REQUESTED** [review]

File: examples/JMX/docker-compose.yaml (hunk 1 above)
Comment (coughman) [reply examples/JMX/docker-compose.yaml]: fixed in latest commit

**Comment by coughman** [comment]
I removed all unrelated envs from all services

File: examples/JMX/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# JMX Docker Compose example
+
+In general, all Kafka services uses the `*JMX_OPTS` environment variable to pass JMX options to the JVM. This compose example has been configured to allow external JMX clients (e.g. jconsole) to connect to the containers' JMX port.
```
Comment (joel-hamill) [inline examples/JMX/README.md]: suggest `...all Kafka services use the..`

Comment (joel-hamill) [inline examples/JMX/README.md]: suggest: `This compose example is configured...`

File: examples/JMX/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# JMX Docker Compose example
+
+In general, all Kafka services uses the `*JMX_OPTS` environment variable to pass JMX options to the JVM. This compose example has been configured to allow external JMX clients (e.g. jconsole) to connect to the containers' JMX port. 
+
+If you log into the container you will see this environment variable being set. This can cause issues if you are running kafka CLI commands locally within the container. To work around this you would need to unset the `*JMX_OPTS` environment variable. For instance for the kafka container, you need to unset the `KAFKA_JMX_OPTS` before you can run any kafka CLI command:
```
Comment (joel-hamill) [inline examples/JMX/README.md]: suggest: `If you are running Kafka CLI commands locally within the container, you must unset the `*JMX_OPTS` environment variable.`

**Review by joel-hamill — APPROVED** [review]
grammar fixes

**Comment by coughman** [comment]
incorporated changes suggested by @joel-hamill

File: examples/JMX/docker-compose.yaml (hunk 2 above)
Comment (coughman) [reply examples/JMX/docker-compose.yaml]: removed KAFKA_REST* env

File: examples/JMX/docker-compose.yaml (hunk 3 above)
Comment (coughman) [reply examples/JMX/docker-compose.yaml]: unfortunately KAFKAREST_JMX_OPTS is used in kafka-rest repo:
https://github.com/confluentinc/kafka-rest/blob/1cb07d62871d8eb24cfeb5939bbe4fc5aa308037/bin/kafka-rest-run-class#L43-L45

File: examples/JMX/docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      - schema-registry
+    ports:
+      - "8082:8082"
+      - "58082:58082"
+    environment:
+      KAFKA_REST_ZOOKEEPER_CONNECT: "zookeeper:2181"
+      KAFKA_REST_BOOTSTRAP_SERVERS: "PLAINTEXT://kafka:9092"
+      KAFKA_REST_SCHEMA_REGISTRY_URL: "schema-registry:8081"
+      KAFKA_REST_HOST_NAME: "rest-proxy"
+      KAFKAREST_JMX_OPTS: "-Dcom.sun.management.jmxremote=true -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.rmi.port=58082 -Dcom.sun.management.jmxremote.port=58082 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.local.only=false -Djava.rmi.server.hostname=0.0.0.0"
```
Comment (lostick) [inline examples/JMX/docker-compose.yaml]: Shouldn't it be `KAFKA_REST_JMX_OPTS` ?

**Pull request closed without merging**